pages/_app.js

```diff
@@ -2,23 +2,50 @@ import GlobalStyles from "../styles/GlobalStyles";
 import Header from "../components/Header";
 import Navigation from "../components/Navigation";
 import {Jost} from "@next/font/google";
-/* import { Main } from "next/document"; */
+import {useState} from "react";
+import AppContext from "../components/AppContext";
+import {useLoadScript} from "@react-google-maps/api";
 
 const jost = Jost({
   subsets: ["latin"],
   weight: "300",
 });
 
+const libraries = ["places", "geometry"];
+
 function MyApp({Component, pageProps}) {
+  const [status, setStatus] = useState(null);
+  const [center, setCenter] = useState(null);
+  const [clickedMarker, setClickedMarker] = useState(null);
+  const [lawyers, setLawyers] = useState([]);
+  const {isLoaded} = useLoadScript({
+    googleMapsApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY,
+    libraries,
+  });
+
   return (
-    <>
-      <main className={jost.className}>
-        <GlobalStyles />
-        <Header />
-        <Component {...pageProps} />
-        <Navigation />
-      </main>
-    </>
+    <AppContext.Provider
+      value={{
+        status,
+        setStatus,
+        center,
+        setCenter,
+        clickedMarker,
```

Comment:
If I understand it correctly, `center` should store the location the map is centered around.
I would prefer to call it `location` or `searchLocation` instead.